datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxHorizontalScalingService.java

```diff
@@ -99,46 +123,86 @@ public String triggerScalingFlow(SdxCluster sdxCluster, StackScaleV4Request scal
+        if (YARN == stackResponse.getCloudPlatform()) {
+            throw new BadRequestException("The Data lake horizontal scaling is not supported on YCloud");
+        }
+        String nodeRequestValid = isNodeRequestValid(stackResponse, request);
+        if (StringUtils.isNotEmpty(nodeRequestValid)) {
+            throw new BadRequestException(nodeRequestValid);
         }
     }
 
     private String isNodeRequestValid(StackV4Response stack, DatalakeHorizontalScaleRequest request) {
         if (request.getDesiredCount() < 0) {
             LOGGER.warn("Negative nodeCount was requested.");
-            return "Negative nodeCount is not accepted. If you want to downsacle your cluster please use lower number than the acutal node count";
+            return "Negative nodeCount is not accepted. If you want to downscale your Data lake please use lower number than the actual node count";
         }
         String target = request.getGroup();
-        boolean hostGroupNotExists = stack.getInstanceGroups().stream()
-                .map(InstanceGroupV4Response::getName)
-                .noneMatch(target::equalsIgnoreCase);
-        if (hostGroupNotExists) {
+        String gatewayInstanceName = DatalakeInstanceGroupScalingDetails.GATEWAY.getName().toLowerCase(Locale.ROOT);
+        Map<String, Set<InstanceMetaDataV4Response>> instanceMap = getInstanceMap(stack, target, gatewayInstanceName);
+        if (!instanceMap.containsKey(target)) {
             LOGGER.warn("The requested hostgroup name not found!");
-            return String.format("The requested hostgroup name not found! The requested host group name is %s", target);
+            return String.format("The requested hostgroup name not found! The requested hostgroup name is %s", target);
+        }
+        if (!instanceMap.containsKey(gatewayInstanceName)) {
+            LOGGER.error("Gateway node is not found in the Stack Instance Map. The horizontal scale cannot be performed while the CM is not " +
+                    "available StackId: {}", stack.getId());
+            List<String> instanceGroupList = stack.getInstanceGroups().stream().map(InstanceGroupV4Base::getName).toList();
+            return String.format("The Gateway instance not found. The horizontal scale cannot be performed while the CM is not reachable" +
+                    ", found instances: %s", instanceGroupList);
+        }
+        List<InstanceMetaDataV4Response> gatewayMetaData = getGatewayHealthyMetadata(gatewayInstanceName, instanceMap);
+        if (!gatewayMetaData.isEmpty()) {
+            LOGGER.warn("Gateway(CM node) is not healthy. Can not start horizontal scaling");
+            List<InstanceStatus> instanceStatuses = gatewayMetaData.stream().map(InstanceMetaDataV4Response::getInstanceStatus).toList();
+            return String.format("Gateway instances are not healthy. Please repair the node and retry horizontal scaling Actual Gateway statuses: %s",
+                    instanceStatuses);
```

Comment:
This error will be thrown even if the gateway node that is down is not the node where CM is running. Is my understanding correct?

Reply:
good. Thanks for the calrification.